### DISCUSSION TITLE
Toggle `loading` using Javascript.

### DISCUSSION THREAD

**Discussion by johncarter-** [livewire/livewire#2005] [1. Help]
I have a some `wire.loading.class="opacity-50 pointer-events-none"` attributes setup on some DOM elements.

I want to make `loading` to be true from a javascript event. Specifically I am using stripe.js with a form submission:

```js
var form = document.getElementById("stripe-payment-form");
form.addEventListener("submit", function(event) {
    event.preventDefault();

    // TODO Set livewire loading state to true.
    //Livewire.emit('loading'); // Nada.

    stripe
        .confirmCardPayment('{{ $paymentIntentClientSecret }}', {

        //... yada
        //... yada
```

I thought I might be able to do `Livewire.emit('loading')`, but that doesn't really cut the mustard. Unless there is a away I can set loading true from the component class?

 Any ideas?

**Comment by joshhanley** [comment]
Hmm I don't think there is a way to hook into that currently.

You could just use alpine to apply those classes to that element before the stripe call happens and then remove them again once it has finished.

Hope this helps!

**Comment by johncarter-** [comment]
Thanks for saving me any more time source diving.

I will come up with a different solution.
